frontend/taipy-gui/src/components/Taipy/ProgressBar.tsx

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2021-2024 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+import * as React from "react";
+import { useEffect, useState } from "react";
+import CircularProgress from "@mui/material/CircularProgress";
+import LinearProgress from "@mui/material/LinearProgress";
+import Box from "@mui/material/Box";
+
+interface ProgressBarProps {
+    linear?: boolean;
+    //showProgress?: boolean;
+}
+
+const Progress = (props: ProgressBarProps) => {
+    const { linear } = props;
+    const [value, setValue] = useState(false); //By default the circular element will be rendered. If the user declares the `linear` param as true, then the linear element will be rendered otherwise circular.
+
+    useEffect(() => {
+        setValue((val) => {
+            if (props.linear !== undefined) return props.linear;
+            return val;
+        });
+    }, [props.linear, linear]);
+
+    return (
+        <>
+            {value == true ? (
+                <Box sx={{ width: "100%" }}>
```

Comment:
I suppose we want to be able to have 2 progress bars side by side without layout 
Maybe use a span instead of a Box/div ?

Reply:
Actually I'm not able to understand this completely. Like are you saying about that if the user wants to have 2 or more progress bars, for that I've to change, right?

Reply:
yes if the declaration is <|progress|><|progress|>, the user wants to see 2 progress bars side by side

Reply:
> I suppose we want to be able to have 2 progress bars side by side without layout Maybe use a span instead of a Box/div ?

But, in the [official doc](https://mui.com/material-ui/react-progress/) of MUi, it's given that we've to use `<Box>`. Will it render the same element if we use `<Span>` instead?

Reply:
You don't need a <span> I would say.
Simply render the <Box> as is (without the useless fragment) and it should be ok.
You could easily test that in a React-supported Codepen, creating two of these next to each other...

Reply:
Ig my current PR is having a lot of bugs😢I'll create a new one instead and add the necessary changes👍🏼Apologies for inconvenience

Reply:
Got it.

Reply:
> Please make sure you read and apply the comments we've added. Thanks a lot!

Sure, I'll definitely add the changes suggested here👍🏼